DB 702: load short colnames and put long names in error reports

bsweger opened the pull request:
This is the next piece of the DB-702 work:

* re-instates the long column names in the validation error reports
* loads short column names into the validation db and adds short<-->long mapping functions to the interface

jworcestBAH commented:
Merge conflicts seem to need resolving

jworcestBAH commented on dataactvalidator/config/awardFinancialFieldsRaw.csv:
```diff
@@ -1,49 +1,50 @@
-ObligationsDeliveredOrdersUnpaidTotal_CPE,obligations_delivered_orde_cpe,Optional,numeric,21,"A16, C3"
-ObligationsDeliveredOrdersUnpaidTotal_FYB,obligations_delivered_orde_fyb,Optional,numeric,21,C3
-ObligationsIncurredTotalByAward_CPE,obligations_incurred_byawa_cpe,Optional,numeric,21,C3
+<<<<<<< HEAD
```
Might work better without this line

bsweger replied:
Is that your professional assessment? On it!

jworcestBAH commented on dataactvalidator/config/programActivityFieldsRaw.csv:
```diff
@@ -3,7 +3,7 @@ AgencyIdentifier,agency_identifier,Required,string,3,"A1, A20"
 AvailabilityTypeCode,availability_type_code,Conditional per Validation Rule,string,1,"A1, A21"
 BeginningPeriodOfAvailability,beginning_period_of_availa,Conditional per Validation Rule,string,4,A1
-ByDirectReimbursableFundingSource,by_direct_reimbursable_fun,Required,string,1,B12
+ByDirectReimbursableFundingSource,by_direct_reimbursable_fun,Conditional per Validation Rule,string,1,B12
```
Good idea to change this here too, I didn't think of the raw files.

jworcestBAH commented on dataactvalidator/interfaces/validatorValidationInterface.py:
```diff
@@ -399,6 +403,18 @@ def getRuleSeverityByName(self, ruleSeverityName):
+    def getLongToShortColname(self):
+        """Return a dictionary that maps schema field names to shorter, machine-friendly versions."""
+        query = self.session.query(FileColumn.name, FileColumn.name_short)
+        dict = {row.name:row.name_short for row in query}
```
I'm a bit confused as to how this is working without actually doing the query.

bsweger replied:
It's a dictionary comprehension, similar the list comprehensions we've been using. Basically, it's constructing a dictionary by looping through the query results.

jworcestBAH replied:
Yes that line is clear, my problem is that the query hasn't had something like query.all() done, so it shouldn't have results available.  If sqlalchemy is being overly clever and allowing you to pull a result out of an unexecuted query by executing it, my concern would then become how often the query is executed if you're iterating through it.  Is there an advantage to doing it this way rather than having the previous line end with .all() to make it more explicit?

bsweger replied:
Just pushed that fix. Makes sense to be explicit--good call (and very Pythonic too!)

jworcestBAH commented:
Looks good, just have a question on how the dictionary query is working, and conflicts need resolving.

bsweger commented:
Thanks for the review @jworcestBAH. Fixed the merge conflicts, removed the HEAD, and tried to explain the dictionary thing. Let me know if you have any more questions!

klrBAH commented:
[run tests]

jworcestBAH commented:
Looks good, will merge as soon as Jenkins does it's thing.
